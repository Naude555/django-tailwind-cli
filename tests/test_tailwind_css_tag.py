from typing import Any, Dict, Union

from django.conf import LazySettings
from django.template import engines


def test_tailwind_css_in_production(settings: LazySettings, snapshot: Any):
    settings.DEBUG = False
    template_string = """
    {% load tailwind_cli %}
    {% tailwind_css %}
    """
    assert _render(template_string) == snapshot


def test_tailwind_css_in_devmode(settings: LazySettings, snapshot: Any):
    settings.DEBUG = False
    template_string = """
    {% load tailwind_cli %}
    {% tailwind_css %}
    """
    assert _render(template_string) == snapshot


def _render(text: str, context: Union[Dict[str, Any], None] = None):
    if context is None:
        context = {}
    template = engines["django"].from_string(text)
    return template.render(context or {})
