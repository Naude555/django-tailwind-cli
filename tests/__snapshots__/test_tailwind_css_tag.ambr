# serializer version: 1
# name: test_tailwind_css_in_devmode
  '''
  
          
          
  
  
    <link rel="stylesheet" href="/static/css/styles.css">
  
  
          
  '''
# ---
# name: test_tailwind_css_in_production
  '''
  
          
          
  
  
    <link rel="preload" href="/static/css/styles.css" as="style">
    <link rel="stylesheet" href="/static/css/styles.css">
  
  
          
  '''
# ---
